Enable /bigobj on MSVC by default

DavidFair opened the pull request:
**Description of work.**
This enables the bigobj flag for targets where the number of sections in
an obj file exceeds 2^16 for ISISReflectometryTest targets causing the build to fail on MSVC debug. 
This is because the underlying target pulls in Qt and MantidQtWidgetsCommon
pushing us just over the threshold.

This is enabled by Microsoft by default for UWP apps (so has first party
support). The drawback is linkers pre-2005 cannot read from these
objects, something that will not affect us:
https://stackoverflow.com/a/15110915

**To test:**
- Build MantidScientificInterfacesISISReflectometryTestQt5 on MSVC Debug mode

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DavidFair committed:
Enable /bigobj on MSVC by default

This enables the bigobj flag for targets where the number of sections in
an obj file exceeds 2^16 for ISISReflectometryTestTargets. This is
because the underlying target pulls in Qt and MantidQtWidgetsCommon
pushing us just over the threshold.

This is enabled by Microsoft by default for UWP apps (so has first party
support). The drawback is linkers pre-2005 cannot read from these
objects, something that will not affect us:
https://stackoverflow.com/a/15110915

martyngigg commented on buildconfig/CMake/MSVCSetup.cmake:
```diff
@@ -32,17 +32,18 @@ string(REGEX REPLACE "/" "\\\\" THIRD_PARTY_INCLUDE_DIR
 set(CMAKE_CXX_FLAGS
     "${CMAKE_CXX_FLAGS} \
-  /MP /W3 \
+  /MP /W3 /bigobj \
```
I had a feeling that this already existed in a couple of places. Could we remove this now it is included globally: https://github.com/mantidproject/mantid/search?q=bigobj

DavidFair committed:
Drop various /bigobj usages from around codebase

Since we now set this globally we can completely drop these one-off
cases previous developers have added

DavidFair committed:
Fix obselete warning flag

Drops warning flags set of various targets, but I cannot find reference
to in the MSVC docs / anywhere else at all. These are easily fixed by
returning the correct type through the boost python layer

martyngigg reviewed (dismissed):
Thanks for cleaning up those old targets.

martyngigg commented:
`cppcheck` seems broken though!

DavidFair commented:
I've rebased onto the latest main :crossed_fingers:

martyngigg reviewed (approved):
Looks good to me now.
